packages/veritone-react-common/src/components/SearchBar/component.js

```diff
@@ -365,33 +368,49 @@ export class SampleSearchBar extends React.Component {
 
   async getLibraries(auth) {
     if(auth) {
-      return await fetch(`${this.props.api}v3/graphql`, {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json',
-          'Authorization': 'Bearer ' + auth
-        },
-        body: JSON.stringify({query:
-          `query {
-            libraries(limit: 50) {
-              records {
-                id
-                name
-                description
-                coverImageUrl
+      let LIBRARY_OFFSET = 0;
+      let libraries = [];
+      while (true) {
```

Comment:
Using this type of loop presents a HUGE risk of infinitely looping if the service fails to respond correctly in any case. Instead, you should abstract the get library call into a separate function which returns a promise and chain the promise returns until we reach the library limit.